lms/djangoapps/ccx/tasks.py

```diff
@@ -0,0 +1,43 @@
+"""
+Asynchronous tasks for the CCX app.
+"""
+
+from django.dispatch import receiver
+import logging
+
+from opaque_keys import InvalidKeyError
+from ccx_keys.locator import CCXLocator
+from xmodule.modulestore.django import SignalHandler
+from lms import CELERY_APP
+
+from .models import CustomCourseForEdX
+
+log = logging.getLogger("edx.ccx")
+
+
+@receiver(SignalHandler.course_published)
+def course_published_handler(sender, course_key, **kwargs):  # pylint: disable=unused-argument
+    """
+    Consume signals that indicate course published. If course already a CCX, do nothing.
+    """
+    if not isinstance(course_key, CCXLocator):
+        send_ccx_course_published.delay(course_key)
```

Comment:
We've had problems before because the celery message serialization is configured to use JSON and the keys won't work properly. Please send the string serialization, or take some other measure to make sure it works right in JSON.